lib/utils/documentdb.ts

```diff
@@ -462,19 +462,23 @@ export function replaceDocument<T>(
   client: DocumentDb.DocumentClient,
   documentUri: IDocumentDbDocumentUri,
   document: T & DocumentDb.NewDocument,
-  partitionKey: string
+  partitionKey: string,
+  etag?: string
```

Comment:
this should not be optional, as you're replacing an existing document, you should always provide the `etag` of the existing document - I suggest to throw an error on the caller site in case the `etag` is not defined

Reply:
To do that we must patch the typings of RetrievedDocument where etag is optional:

```
/** Represents the common meta data for all DocumentDB objects. */
export interface AbstractMeta extends UniqueId {
    /** The self link. */
    _self: string;

    /** The time the object was created. */
    _ts: number;

    _rid?: string;
    _etag?: string;

    _attachments?: string;
}
```

This because the etag value comes from some RetrievedDocument returned by the find method: 

https://github.com/teamdigitale/digital-citizenship-functions/blob/4836559cf4370a5b2bdb71ba5d8499d1829add2a/lib/utils/documentdb_model.ts#L148

Reply:
I was thinking about something like:

```ts
const doc = findDoc(...);
const currentETag = doc._etag;
if(currentETag === undefined) {
  throw new Error("OMG");
}
client.update(newDoc, doc._etag);

```

Reply:
anoter option is to get rid of this method entirely as everything should be immutable

Reply:
I like this option

Reply:
done